ci-operator/step-registry/baremetalds/e2e/test/baremetalds-e2e-test-commands.sh

```diff
@@ -34,15 +34,49 @@ scp "${SSHOPTS[@]}" /usr/bin/openshift-tests /usr/bin/kubectl "root@${IP}:/usr/l
 
 # Tests execution
 set +e
-if [[ -s "${SHARED_DIR}/test-list" ]]; then
-    echo "### Copying test-list file"
-    scp "${SSHOPTS[@]}" "${SHARED_DIR}/test-list" "root@${IP}:/tmp/test-list"
-    echo "### Running tests"
-    ssh "${SSHOPTS[@]}" "root@${IP}" openshift-tests run "openshift/conformance/parallel" --dry-run \| grep -Ff /tmp/test-list \|openshift-tests run -o /tmp/artifacts/e2e.log --junit-dir /tmp/artifacts/junit -f -
+
+# Test upgrade for workflows that requested it
+if [[ "$RUN_UPGRADE_TEST" == true ]]; then
```

Comment:
The check it's ok, anyhow I'd suggest to follow an approach similar to the one in https://github.com/openshift/release/pull/12959/ (to be as much as possible similar to the ones adopted by the other profiles) where possible, ie making a check directly against `OPENSHIFT_UPGRADE_RELEASE_IMAGE` as in https://github.com/openshift/release/blob/d4f03b5848e88d409f6b0cac7ec205f47530e07b/ci-operator/step-registry/openshift/e2e/test/openshift-e2e-test-commands.sh#L75 (we'll save one var)

Reply:
I thought about this approach, the problem is that `OPENSHIFT_UPGRADE_RELEASE_IMAGE` will always be non-empty since the step specifies it as a dependency, so I've needed a new variable.

From where the `TEST_COMMAND` variable comes from? I didn't see it in any of the templates.

Reply:
Yes, I think that's a sanity check (always expected to be not empty). They've defined `TEST_COMMAND` as a normal env var in the [openshift-e2e-test](https://github.com/openshift/release/blob/e808d50ef535328177564e0f7e4690218e8fc3f1/ci-operator/step-registry/openshift/e2e/test/openshift-e2e-test-ref.yaml#L7).
In our case we've the additional requirement to filter our some of the conformance tests (in the case of installation test), checking the current value of `TEST_COMMAND` could help in crafting properly the various pieces required by `openshift-tests` (the command itself, additional options such as `--to-image` and the filter list)

Reply:
I'd prefer to adopt it if possible in the current PR, so that also the code duplication will be removed. If not, it's ok to address it in another PR.